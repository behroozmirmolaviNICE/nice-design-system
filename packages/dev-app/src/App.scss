@import "~@nice-digital/design-system/scss/nice-design-system";

@import "~@nice-digital/nds-forms/scss/forms";

main {
  background: white;
  box-shadow: 0 0 10px rgba(0, 0, 0, .3);
  margin: 3vw;
  padding: 3vw;
  position: relative;
}
