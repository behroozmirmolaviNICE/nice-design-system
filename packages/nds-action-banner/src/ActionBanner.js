// @flow
import React, { Component } from "react";
import "../scss/action-banner.scss";

type ActionBannerProps = {
	title: string,
	modifier: "subtle",
	children: *,
	cta: Element,
	closeable: boolean
};

const ActionBanner = (props: ActionBannerProps) => {

	let classes = ["action-banner"];

	if(props.modifier)
		classes.push(`action-banner--${props.modifier}`);

	if(props.closeable)
		classes.push("action-banner--closeable");

	return <section className={classes.join(" ")}>
		<div className="action-banner__container">
			<div className="action-banner__inner">
				<div className="action-banner__text">
					<h2 className="action-banner__title">
						{props.title}
					</h2>
					<p className="action-banner__intro">
						{props.children}
					</p>
				</div>
				<div className="action-banner__actions">
					{props.cta}
				</div>
				{props.closeable &&
					<button type="button" className="action-banner__close">
						<span className="icon icon--remove" ara-hidden="true"></span>
						<span className="visually-hidden">Close</span>
					</button>
				}
			</div>
		</div>
	</section>;
};

export default ActionBanner;
