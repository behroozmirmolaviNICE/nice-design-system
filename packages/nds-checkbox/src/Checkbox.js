import React, { Component } from "react";
import PropTypes from "prop-types";
import "../scss/checkbox.scss";

export class Checkbox extends Component {
	render() {
		return <div>Checkbox!!!</div>;
	}
}

Checkbox.propTypes = {};
