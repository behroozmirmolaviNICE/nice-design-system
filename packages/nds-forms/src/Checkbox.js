// @flow
import React, { Component } from "react";
import "../scss/checkbox.scss";

type CheckboxProps = {
	name: string,
	label: string,
	value: string,
	inline: boolean,
};

export default class Checkbox extends Component<CheckboxProps> {

	render() {
		const {inline, name, label, value, ...rest} = this.props;
		const unique = name + "_" + value;
		const classNames = inline ? "checkbox checkbox--inline" : "checkbox";
		return (
			<div className={classNames}>
				<input
					type="checkbox"
					className="checkbox__input"
					id={unique}
					name={name}
					value={value}
					{...rest}
				/>
				<label
					className="checkbox__label"
					htmlFor={unique}>
					{label}
				</label>
			</div>
		);
	}
}
