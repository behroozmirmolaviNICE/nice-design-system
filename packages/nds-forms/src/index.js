import Radio from "./Radio";
import RadioGroup from "./RadioGroup";
import Textarea from "./Textarea";
import Input from "./Input";
import Fieldset from "./Fieldset";

export { Radio, RadioGroup, Textarea, Input, Fieldset };
