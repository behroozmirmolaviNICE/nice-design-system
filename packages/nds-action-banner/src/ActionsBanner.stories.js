import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";

import ActionBanner from "./ActionBanner";

storiesOf("Action banner", module)
	.addDecorator(withKnobs)
	.add("Primary", () => <ActionBanner />);
