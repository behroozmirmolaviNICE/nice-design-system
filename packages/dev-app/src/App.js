import React, { Component } from "react";
import { Radio, RadioGroup, Textarea, Input, Fieldset, Checkbox } from "@nice-digital/nds-forms";

import "./App.scss";

class App extends Component {
	render() {
		return (
			<div>

				<Fieldset legend="Checkboxes">
					<Checkbox />
				</Fieldset>

				<RadioGroup
					legend="Radio buttons (stacked, default)"
					group="group-1"
					hint="This is some hint text you can read">
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..? Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ea eaque explicabo, facere harum necessitatibus nulla optio pariatur rem! Aut blanditiis dolores expedita minima mollitia nostrum rem! Accusamus aliquid, distinctio facilis harum illum magni nemo pariatur provident sequi vel, voluptates voluptatum!"/>
				</RadioGroup>

				<RadioGroup legend="Radio buttons inline" group="group-2" hint="This is some hint text you can read" inline>
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..?"/>
				</RadioGroup>

				<Fieldset legend="Textarea">
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						value="Default value for textarea"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						hint="This is some hint text you can read"
						name="myCommentBox"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						name="mySecondCommentBox"/>
				</Fieldset>

				<Fieldset legend="Inputs">
					<Input
						name="full-name"
						type="text"
						label="Full name"/>
					<Input
						name="email"
						type="email"
						label="Your email address"/>
					<Input
						name="password"
						type="password"
						hint="Your password must contain 4 special characters, 6 numbers, 12 letters and an emoji"
						label="Password"/>
				</Fieldset>

			</div>
		);
	}
}

export default App;
