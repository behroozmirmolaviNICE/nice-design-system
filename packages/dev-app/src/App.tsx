import React, { Component } from "react";
// import Faker from "faker";

import { Checkbox } from "@nice-digital/nds-checkbox";
import "./App.scss";

class App extends Component {
	render() {
		return (
			<main>
				<Checkbox
					label="Email"
					value="email"
					name="contact-preference"
					error={true}
				/>
				<Checkbox
					label="Telephone"
					value="telephone"
					name="contact-preference"
				/>
				<Checkbox
					label="Text Message"
					value="sms"
					name="contact-preference"
					error="You must accept the terms and conditions"
				/>
			</main>
		);
	}
}

export default App;
