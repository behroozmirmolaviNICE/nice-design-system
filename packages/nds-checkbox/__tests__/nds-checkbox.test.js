import React from "react";
import { shallow, mount } from "enzyme";

import { Checkbox } from "./../src/Checkbox";

describe("Checkbox", () => {
	it("renders without crashing", () => {
		const wrapper = shallow(<Checkbox />);
		expect(wrapper).toHaveLength(1);
	});
});
