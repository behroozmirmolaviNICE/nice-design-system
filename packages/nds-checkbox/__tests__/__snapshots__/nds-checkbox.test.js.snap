// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot with hint text supplied 1`] = `
<div
  className="checkbox"
>
  <input
    className="checkbox__input"
    id="contact-preference_email"
    name="contact-preference"
    type="checkbox"
    value="email"
  />
  <label
    className="checkbox__label"
    htmlFor="contact-preference_email"
  >
    Email
  </label>
  <span
    className="checkbox__hint"
  >
    This is a hint to appear.
  </span>
</div>
`;

exports[`Checkbox should match snapshot with some default attributes 1`] = `
<div
  className="checkbox"
>
  <input
    className="checkbox__input"
    id="contact-preference_email"
    name="contact-preference"
    type="checkbox"
    value="email"
  />
  <label
    className="checkbox__label"
    htmlFor="contact-preference_email"
  >
    Email
  </label>
</div>
`;
